back-end/helpers/intent_analyzer.js

```diff
@@ -0,0 +1,56 @@
+export default class IntentAnalyzer{
+    /* What it should result in:
+    [
+        {
+          question: "What is something that Waluigi and Wario often say?",
+          children: {
+            "The brothers Waluigi and Wario have their catchphrase 'waaa'.": 100,
+          }
+        },
+        {
+          question: "Where should I go for a good time and great learning experience?",
+          children: {
+            "You should go to the University of Toronto.": 10,
+            "You should go to the Technology Leadership Initiative.": 90
+          }
+        }, 
+        {
+          question: "Hey BestFlow, what time is it?",
+          children: {
+            "It's time for you to get a watch, would you like to take a look at our product?": 10,
+            "Showtime showtime what I'm John Laurens in the place to be.": 20,
+            "Summertime! School's out, scream and shout.": 70
+          }
+        }
+    ]
+      */
+
+    static async formatIntents(intents) { // Formats the mongoose Model into desired array for IntentLister to use
+      if (Object.keys(intents).length == 0) {
+        return {};
+      }
+
+      var percentageMap;
```

Comment:
Also another javascript learning moment, there are such nuanced differences between var, const, and let, what is the best practice in declaring variables? All of the articles I read are very hard to understand

Reply:
`const` is for defining a variable whose value will never change (like a fixed parameter, or a fixed set of fake intents), and `let` is used if the value will change. `const` is more restrictive but preferred to make code maintenance easier (eliminates chance of fixed values being changed accidentally), and maybe it has memory benefits? Not sure.

I think `var` is similar to let except instead of being scoped to the immediate block of code (i.e. enclosed by { }), `var` is function-scoped. So there are cases where `var` may be more convenient, but only if it's first defined in a nested block for whatever reason. Most guides online seem to say not to use `var` ever as it is an old keyword that should mostly be replaced by either let or const.

Summary: use `const` until you can't, then use `let`. If anyone can clarify on the `var` stuff please add on.

Reply:
I will then be changing my vars to consts, tyvm good sir [tips fedora]